projects/plugins/jetpack/extensions/blocks/dialogue/edit.js

```diff
@@ -98,25 +86,32 @@ export default function DialogueEdit( {
 		// or when there is not a dialogue pre block.
 		// or when there are not particpants,
 		// or there is not conversation bridge.
-		if (
-			participantSlug ||
-			! prevBlock ||
-			! participants?.length ||
-			! conversationBridge
-		) {
+		if ( participant || ! prevBlock || ! participants?.length || ! conversationBridge ) {
 			return;
 		}
 
-		const nextParticipantSlug = conversationBridge.getNextParticipantSlug(
-			prevBlock?.attributes?.participantSlug
+		const nextParticipant = conversationBridge.getNextParticipant(
+			prevBlock?.attributes?.participant?.slug
 		);
 
 		setAttributes( {
 			...( prevBlock?.attributes || {} ),
-			participantSlug: nextParticipantSlug,
+			participant: nextParticipant,
 			content: '',
 		} );
-	}, [ participantSlug, participants, prevBlock, setAttributes, conversationBridge ] );
+	}, [ participant, participants, prevBlock, setAttributes, conversationBridge ] );
+
+	// Update dialog participant with conversation participant changes.
+	useEffect( () => {
+		if ( ! isEqual( conversationParticipant, participant ) ) {
+			setAttributes( { participant: conversationParticipant } );
+		}
+	}, [ conversationParticipant, participant, setAttributes ] );
+
+	// Update dialogue timestamp setting from parent conversation.
+	useEffect( () => {
+		setAttributes( { showTimestamp: showConversationTimestamps } );
+	}, [ showConversationTimestamps, setAttributes ] );
```

Comment:
These effects make sure that the participant and timestamp settings are copied into the dialogue block attributes, when they are changed in the parent conversation context.